### PR TITLE
[iOS] setButtons support UIBarButtonSystemItem and buttonColor support for iOS 

### PR DESCRIPTION
**Pull request by freesteel**
#[iOS] setButtons support UIBarButtonSystemItem and buttonColor support for iOS.

Create system style bar button by set   _buttonSystemItem_

`this.props.navigator.setButtons({
		rightButtons: [{
			buttonColor: 'Orange',
		        buttonSystemItem: 'add',
		}]
})`

 values  refer to<React/RCTNavItemManager.h>

### CODE REVIEW

**Comment by guyca** [comment]
Hey @freesteel 
Thanks for the contribution. Can you please give more details as to what this PR is trying to do? screen shots would be helpful as well. Thanks

**Comment by freesteel** [comment]
d2080ae1-e477-44b5-b810-86ed92a9de0e

Implement by calling the method:
`initWithBarButtonSystemItem: target: action:`
`setTintColor:`

**Comment by guyca** [comment]
Hey @freesteel 
I'm not sure I understand, how does the button you've added different from the buttons currently added by rnn as deescribed here: https://wix.github.io/react-native-navigation/#/adding-buttons-to-the-navigator

**Comment by freesteel** [comment]
Hey @guyca 
I don't need to make icon(image), when using system BarButtonSystemItem.

Sorry,my English is not so good。

**Comment by carlosroberto555** [comment]
Hey @freesteel 
I think I understood your idea,
You want to use a system icon on button, and not a custom that you loading.
I hope I have helped.
@guyca

**Comment by freesteel** [comment]
Hey @carlosroberto555 

That you said is What I want to express

File: ios/RCCNavigationController.m [hunk 1]
```diff
@@ -326,8 +327,20 @@ -(void)setButtons:(NSArray*)buttons viewController:(UIViewController*)viewContro
     id icon = button[@"icon"];
     if (icon) iconImage = [RCTConvert UIImage:icon];
     
+    id systemIcon =button[@"buttonSystemItem"];
```
Comment (eliperkins) [inline ios/RCCNavigationController.m]: Can you add a space here after the `=`?

File: ios/RCCNavigationController.m [hunk 2]
```diff
@@ -326,8 +327,20 @@ -(void)setButtons:(NSArray*)buttons viewController:(UIViewController*)viewContro
...
+    UIBarButtonSystemItem systemItemIcon=NSNotFound;
```
Comment (eliperkins) [inline ios/RCCNavigationController.m]: And spaces around the `=` here again.

File: ios/RCCNavigationController.m [hunk 3]
```diff
@@ -326,8 +327,20 @@ -(void)setButtons:(NSArray*)buttons viewController:(UIViewController*)viewContro
...
+    if (systemIcon) systemItemIcon =[RCTConvert UIBarButtonSystemItem:button[@"buttonSystemItem"]];
```
Comment (eliperkins) [inline ios/RCCNavigationController.m]: One more time with the spaces around `=`.

File: ios/RCCNavigationController.m [hunk 4]
```diff
@@ -326,8 +327,20 @@ -(void)setButtons:(NSArray*)buttons viewController:(UIViewController*)viewContro
...
+
+    id buttonColorTag = button[@"buttonColor"];
+    UIColor *buttonColor = nil;
+    if (buttonColorTag) buttonColor= [RCTConvert UIColor:buttonColorTag];
```
Comment (eliperkins) [inline ios/RCCNavigationController.m]: And here as well.

File: ios/RCCNavigationController.m [hunk 5]
```diff
@@ -326,8 +327,20 @@ -(void)setButtons:(NSArray*)buttons viewController:(UIViewController*)viewContro
...
+
+
     UIBarButtonItem *barButtonItem;
-    if (iconImage)
+    if (systemItemIcon!=NSNotFound) {
```
Comment (eliperkins) [inline ios/RCCNavigationController.m]: Can you add spaces around `!=` here too?

Comment (eliperkins) [inline ios/RCCNavigationController.m]: Nice catch in noticing that [`-[RCTConvert UIBarButtonSystemItem:]` returns `NSNotFound` by default](https://github.com/facebook/react-native/blob/40fdd6d91c578934b9e1035a167888032fb85d28/React/Views/RCTNavItemManager.m#L42)!

File: ios/RCCNavigationController.m [hunk 6]
```diff
@@ -326,8 +327,20 @@ -(void)setButtons:(NSArray*)buttons viewController:(UIViewController*)viewContro
...
+        barButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:systemItemIcon  target:self action:@selector(onButtonPress:)];
```
Comment (eliperkins) [inline ios/RCCNavigationController.m]: It looks like there are two spaces between `systemItemIcon` and `target`. Do you mind making this just one space?

**Comment by ziyafenn** [comment]
Thanks @freesteel !

**Pull request closed without merging**